Change optional array from deprecated to removed

D0CTOR-ZED opened the pull request:
Change the description of the `optional` array to state it has been removed as https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/patches/minecraft/net/minecraft/tags/Tag.java.patch no longer supports it.  Leave the reference to it in the doc to direct users to the vanilla method.

D0CTOR-ZED committed:
Change optional array from depreciated to removed

Change the description of the `optional` array to state it has been removed as https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/patches/minecraft/net/minecraft/tags/Tag.java.patch no longer supports it.  Leave the reference to it in the doc to direct users to the vanilla method.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -11,10 +11,9 @@ For example, to add your own mod's saplings to the Vanilla sapling tag, you woul
 See the [Vanilla wiki][tags] for a description of the base syntax.
 
-Forge provides two extensions on the Vanilla syntax:
+Forge extensions on the Vanilla syntax:
```
Not really a proper way to define it. This can be removed in favor of just listing it as part of the description.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -11,10 +11,9 @@ For example, to add your own mod's saplings to the Vanilla sapling tag, you woul
 
-* You may declare an `optional` array of the same format as the `values` array, but any values listed here that are not present will not cause the tag loading to error.
-This has been deprecated in favor of the vanilla method of specifying optional tag values.
+* The `optional` array was depreciated and has been removed in favor of the vanilla method of specifying optional tag values.
```
This was removed back in 1.17, so apologies on missing it. But instead of mentioning it's removal, it should instead mention the correct way to declare optional tag values.

Additionally, `depreciated` is spelled incorrectly. It is `deprecated`.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -11,10 +11,9 @@ For example, to add your own mod's saplings to the Vanilla sapling tag, you woul
-* You may declare an `optional` array of the same format as the `values` array, but any values listed here that are not present will not cause the tag loading to error.
-This has been deprecated in favor of the vanilla method of specifying optional tag values.
+* The `optional` array was depreciated and has been removed in favor of the vanilla method of specifying optional tag values.
 * You may declare a `remove` array of the same format as the `values` array. Any values listed here will be removed from the tag. This acts as a finer grained version of the Vanilla `replace` option.
```
This would be where you mention a forge added extension since its singular now.

ChampionAsh5357 reviewed (changes requested):
Please leave examples by pointing to the exact code references instead of a patch that hasn't changed in relation to what you are describing. For example the [1.16.x version](https://github.com/MinecraftForge/MinecraftForge/blob/1.16.x/src/main/java/net/minecraftforge/common/ForgeHooks.java#L1163-L1191) compared to the [1.18.x version](https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/src/main/java/net/minecraftforge/common/ForgeHooks.java#L1071-L1086).

D0CTOR-ZED committed:
Update tags.md

Move optional tag explanation into vanilla tag description.  Change forge extension to singular.

D0CTOR-ZED commented:
Not sure if I've implemented the edits correctly.  I'm learning github as I go here, so if there is something else I should have done, please let me know.  It shows where I updated above, so hopefully I'm on track.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -9,12 +9,11 @@ Tags are declared in your mod's [datapack][datapack]. For example, `/data/<modid
 Similarly, you may append to or override tags declared in other domains, such as Vanilla, by declaring your own JSONs.
 For example, to add your own mod's saplings to the Vanilla sapling tag, you would specify it in `/data/minecraft/tags/blocks/saplings.json`, and Vanilla will merge everything into one tag at reload, if the `replace` option is false.
 If `replace` is true, then all entries before the json specifying `replace` will be removed.
+Entries which may not be present, such as resources from other mods, can be included using a compound tag that uses an id string and required boolean set to false.
```
Reading this does not make much sense. Compound tags are not a thing in JSON. There is an alternative representation in JSON objects, but they are not the same. Provide an example when stating a case.

Additionally, when saying that making something optional, you should clearly express that point. (Ex. `If a tag entry should not cause the tag to fail to load...`)

D0CTOR-ZED replied:
Interesting.  I used the term since it was the alt text assigned to the image that labeled the relevant part in the vanilla wiki. I'll do some reading regarding the json standard and see what the proper term is.

ChampionAsh5357 replied:
Yes, data is stored either using the NBT format (to which objects are represented as Compound Tags) or JSON format (to which objects are represented as JSON Objects). Tags can be serialized in both formats, though the one datapacks use is the JSON format.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -9,12 +9,11 @@ Tags are declared in your mod's [datapack][datapack]. For example, `/data/<modid
-Forge provides two extensions on the Vanilla syntax:
+Forge extension on the Vanilla syntax:
 
-* You may declare an `optional` array of the same format as the `values` array, but any values listed here that are not present will not cause the tag loading to error.
-This has been deprecated in favor of the vanilla method of specifying optional tag values.
 * You may declare a `remove` array of the same format as the `values` array. Any values listed here will be removed from the tag. This acts as a finer grained version of the Vanilla `replace` option.
```
There's only one extension, just make it a single sentence.

D0CTOR-ZED committed:
Update tags.md

Adjust text based on feedback (https://github.com/MinecraftForge/Documentation/pull/401/files#r791868995)

D0CTOR-ZED committed:
Update tags.md

Markup keywords.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -9,13 +9,22 @@ Tags are declared in your mod's [datapack][datapack]. For example, `/data/<modid
 For example, to add your own mod's saplings to the Vanilla sapling tag, you would specify it in `/data/minecraft/tags/blocks/saplings.json`, and Vanilla will merge everything into one tag at reload, if the `replace` option is false.
 If `replace` is true, then all entries before the json specifying `replace` will be removed.
-See the [Vanilla wiki][tags] for a description of the base syntax.
+Values listed that are not present will cause the tag to error unless the value is listed using an `id` string and `required` boolean set to false as in the following example.
```
You mention `...as in the following example` above the json as well. It should either be removed from the sentence or removed from above the json and instead ending in `..as in the following example:`

ChampionAsh5357 commented on docs/utilities/tags.md:
````diff
@@ -9,13 +9,22 @@ Tags are declared in your mod's [datapack][datapack]. For example, `/data/<modid
-Forge provides two extensions on the Vanilla syntax:
+As an example:
+```json
+{ "replace": false,
+  "values": [
+    "minecraft:gold_ingot",
+    "mymod:my_ingot",
+    { "id": "othermod:ingot_other", "required": false }
+  ]
+}
````
JSON should be formatted in pretty printing since that's how json objects are generated.

Example:
```json
{
  "key1": "value1",
  "key2": [
    "list1",
    "list2",
    {
      "innerkey1": "innervalue1",
      "innerkey2": "innervalue2"
    }
  ]
}
```

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -9,13 +9,22 @@ Tags are declared in your mod's [datapack][datapack]. For example, `/data/<modid
-This has been deprecated in favor of the vanilla method of specifying optional tag values.
-* You may declare a `remove` array of the same format as the `values` array. Any values listed here will be removed from the tag. This acts as a finer grained version of the Vanilla `replace` option.
+See the [Vanilla wiki][tags] for a description of the base syntax.
+  
```
New line, not double space.

D0CTOR-ZED committed:
Update tags.md

Make minor formatting change.  Make JSON pretty.

ChampionAsh5357 commented on docs/utilities/tags.md:
```diff
@@ -9,13 +9,24 @@ Tags are declared in your mod's [datapack][datapack]. For example, `/data/<modid
-* You may declare an `optional` array of the same format as the `values` array, but any values listed here that are not present will not cause the tag loading to error.
-This has been deprecated in favor of the vanilla method of specifying optional tag values.
-* You may declare a `remove` array of the same format as the `values` array. Any values listed here will be removed from the tag. This acts as a finer grained version of the Vanilla `replace` option.
+See the [Vanilla wiki][tags] for a description of the base syntax.
+There is also a Forge extension on the Vanilla syntax. You may declare a `remove` array of the same format as the `values` array. Any values listed here will be removed from the tag. This acts as a finer grained version of the Vanilla `replace` option.
```
New line mean replace the double space with a new line, not just remove the double space altogether without adding the new line.

D0CTOR-ZED replied:
Instructions were unclear.  Had removed this double spacing: https://github.com/D0CTOR-ZED/Documentation/commit/7f46947f2c519b700fbbeba25eb8b11352e63cc9#diff-8c6dfe5a26f71f428c40a4e8e37dbfa99968d6ee1ad2d6e224f14c112d345a19L26

Had interpreted this, https://github.com/MinecraftForge/Documentation/pull/401#discussion_r791869271 , as meaning that you wanted the parts together.  Re-separated them.  Assuming this is the new line you wanted.

D0CTOR-ZED committed:
Update tags.md

D0CTOR-ZED committed:
Update tags.md
